### PR TITLE
Build fix

### PR DESCRIPTION
**Pull request by microbit-matt-hillsdon**
As per #230, the last two releases are broken due to a babel-loader / babel mismatch.

They've still built and published as the build script doesn't correctly check errors.

This PR fixes the build script to fail on errors (tested with the existing error) and then fixes the build by reverting babel-loader to v7. I haven't tried an upgrade to v8.

If you ever get a chance to rethink the build scripts it might be worth seeing if you can use webpack's cli directly now, rather than the custom script which is more error prone. Even if you want to fix the babel conflict differently, please consider at least the changes from dc0f9cb413ae3af9476cccfadbd7c97eb18b2150 which should ensure failing builds can't publish packages.

Happy to add the version bumps suggested below if this is going straight through to a release. Let me know.

- [ ] `package.json:version` has been updated with `npm version patch` (or `major/minor` as appropriate)
- [ ] `@brainhubeu/react-carousel` version has been updated in `docs-www/package.json` to the same which is in `package.json:version`

### CODE REVIEW

**Comment by microbit-matt-hillsdon** [comment]
Build is failing in the publish GH pages step querying GH. Not clear why (the response from that URL does have a state field for now). Could be a rate limit or other misc GitHub error response. I can't trigger a retry from CircleCI.

**Comment by piotr-s-brainhub** [comment]
@microbit-matt-hillsdon 

I reverted these upgrades and it should be fixed in 1.10.54

File: tools/build.js [hunk 1]
```diff
@@ -14,24 +14,23 @@ console.log(chalkProcessing('Generating minified bundle. This will take a moment
...
     console.log(chalkError(error));
-    return 1;
+    process.exit(1);
   }
 
   const jsonStats = stats.toJson();
 
-  if (jsonStats.hasErrors) {
-    return jsonStats.errors.map(error => console.log(chalkError(error)));
+  if (stats.hasErrors()) {
```
Comment (piotr-s-brainhub) [inline tools/build.js]: why `stats` instead of `jsonStats`?

Comment (piotr-s-brainhub) [reply tools/build.js]: @microbit-matt-hillsdon ?

Comment (microbit-matt-hillsdon) [reply tools/build.js]: Sorry for the delay.

For me, `jsonStats` just has `errors` and `warnings` fields, and doesn't have a `hasErrors` field. The stats object has `hasErrors()` though. I guess you could check errors.length, but it seemed reasonable to use stats as per the webpack docs suggestion for checking for errors: 

https://webpack.js.org/api/node/#error-handling

Now you've fixed the build separately, I could rework this into a PR that just has the build changes (and maybe a version bump?), if that's helpful, or you can just grab the bits you want and close.

**Comment by piotr-s-brainhub** [comment]
@microbit-matt-hillsdon 

The deployment to the testing environment works only for the `brainhubeu` branchs so that's way the build failed.

I opened this PR https://github.com/brainhubeu/react-carousel/pull/242 instead.

**Pull request merged**